packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,231 @@
+    activeBorderColor: getIn(theme.colors, 'action.border.primary.active'),
+    focusColor: getIn(theme.colors, 'action.background.primary.focus'),
+    focusBorderColor: getIn(theme.colors, 'action.border.primary.focus'),
+    focusRingColor: getIn(theme.colors, 'brand.primary.400'),
+    borderWidth: makeBorderSize(theme.border.width.thin),
+    borderRadius: makeBorderSize(theme.border.radius.small),
+  };
+
+  if (isDisabled) {
+    const disabledColor = getIn(theme.colors, 'action.background.primary.disabled');
+    const disabledBorderColor = getIn(theme.colors, 'action.border.primary.disabled');
+    props.iconColor = 'action.icon.primary.disabled';
+    props.textColor = 'action.text.primary.disabled';
+    props.defaultColor = disabledColor;
+    props.defaultBorderColor = disabledBorderColor;
+    props.hoverColor = disabledColor;
+    props.hoverBorderColor = disabledBorderColor;
+    props.activeColor = disabledColor;
+    props.activeBorderColor = disabledBorderColor;
+    props.focusColor = disabledColor;
+    props.focusBorderColor = disabledBorderColor;
+  }
+
+  return props;
+};
+
+const BaseButton = ({
+  variant = 'primary',
+  intent,
+  contrast = 'low',
+  size = 'medium',
+  icon: Icon,
+  iconPosition = 'left',
+  isDisabled = false,
+  isFullWidth = false,
+  onClick,
+  type = 'button',
+  children,
+}: BaseButtonProps): ReactElement => {
+  const { theme, platform } = useTheme();
+  const buttonTypographyTokens = buttonTypography[platform];
+  if (!Icon && !children?.trim()) {
+    throw new Error(
+      `[Blade: BaseButton]: At least one of icon or text is required to render a button.`,
+    );
+  }
+  const {
+    activeBorderColor,
+    activeColor,
+    defaultBorderColor,
+    defaultColor,
+    minHeight,
+    spacing,
+    focusBorderColor,
+    focusColor,
+    focusRingColor,
+    fontSize,
+    hoverBorderColor,
+    hoverColor,
+    iconColor,
+    iconSize,
+    iconSpacing,
+    lineHeight,
+    text,
+    textColor,
+    borderWidth,
+    borderRadius,
+  } = getProps({
+    buttonTypographyTokens,
+    children,
+    isDisabled,
+    size,
+    theme,
+  });
+  console.log('unused props', {
+    variant,
+    intent,
+    contrast,
+  });
+  return (
+    <StyledBaseButton
+      activeBorderColor={activeBorderColor}
+      activeColor={activeColor}
+      defaultBorderColor={defaultBorderColor}
+      minHeight={minHeight}
+      spacing={spacing}
+      defaultColor={defaultColor}
+      disabled={isDisabled}
+      focusBorderColor={focusBorderColor}
+      focusColor={focusColor}
+      focusRingColor={focusRingColor}
+      hoverBorderColor={hoverBorderColor}
+      hoverColor={hoverColor}
+      isFullWidth={isFullWidth}
+      onClick={onClick}
+      type={type}
+      borderWidth={borderWidth}
+      borderRadius={borderRadius}
+    >
+      {Icon && iconPosition == 'left' ? <Icon size={iconSize} color={iconColor} /> : null}
+      {text && (
```

Comment:
nit: follow one pattern?

above you're using ternary operator for conditional rendering, here you're using && operator?